packages/core/src/common/document-manager.ts

```diff
@@ -0,0 +1,46 @@
+import { GlintConfig } from '../config/config.js';
+import TransformManager from './transform-manager.js';
+
+export class DocumentManager {
```

Comment:
I'm a little unclear on what the purpose of this class is (especially next to the simiarly-named `DocumentCache`) — is the state here something that anything other than the language server's internals should be aware of?